### PR TITLE
Add visual loading.. spinner to place order

### PR DESCRIPTION
**Pull request by JustinElst**
## Summary
It can take some time to process the place order wire, all the while the user does not know that a process is running.
This will add a loading spinner for the place order event.

**Tested scenarios**
- Card payments w/wo 3DS2 happy/unhappy flows
- Klarna redirect with success and cancellation flows
- Google Pay wallet payment

### CODE REVIEW

File: view/frontend/templates/payment/model/adyen-payment-method.phtml [hunk 1]
```diff
@@ -355,6 +356,8 @@ $billingAddress = $block->getQuoteBillingAddress();
                     self.handleAdyenResult(paymentStatus);
                 }).catch(() => {
                     console.log('Error occurred during order placement')
+                }).finish(() =>  {
+                    window.dispatchEvent(new Event('magewire:loader:done')
```
Comment (candemiralp) [inline view/frontend/templates/payment/model/adyen-payment-method.phtml:360]:
```suggestion
                    window.dispatchEvent(new Event('magewire:loader:done'));
```
Missing closing parenthesis causes exception.

File: view/frontend/templates/payment/model/adyen-payment-method.phtml [hunk 2]
```diff
@@ -355,6 +356,8 @@ $billingAddress = $block->getQuoteBillingAddress();
                     self.handleAdyenResult(paymentStatus);
                 }).catch(() => {
                     console.log('Error occurred during order placement')
+                }).finish(() =>  {
```
Comment (candemiralp) [inline view/frontend/templates/payment/model/adyen-payment-method.phtml:359]:
```
checkout/:784 Uncaught (in promise) TypeError: wire.placeOrder(...).then(...).catch(...).finish is not a function
    at componentHandler.placeOrder (checkout/:784:20)
    at Object.placeOrder (checkout/:582:30)
    at checkout/:3560:70
    at async Proxy.onClick (checkout/:4102:25)
```

**Review by candemiralp — CHANGES_REQUESTED** [review]

**Comment by candemiralp** [comment]
Hello @JustinElst,

Thank you for your contribution. I've left a comment about an exception message that I got on my checkout. I've tried with 3DS2 and non 3DS2 card payments. Could you please elaborate on the issue?

Best Regards,
Can

File: view/frontend/templates/payment/model/adyen-payment-method.phtml (hunk 1 above)
Comment (JustinElst) [reply view/frontend/templates/payment/model/adyen-payment-method.phtml]: This is fixed :) thanks.

**Comment by JustinElst** [comment]
Thanks for your comments.
When a customer clicks the placeOrder button, a wire.placeOrder is triggered. This could take some time to in our case it does a round trip to the erp system to get a order-number. But in other cases with a slow internet or server this could take some time.

**Comment by candemiralp** [comment]
**FYI CODEOWNERS**, the tests have been deployed locally and PR has passed the tests successfully.

**Review by candemiralp — APPROVED** [review]

**Comment by candemiralp** [comment]
Hello @JustinElst,

Thank you for your contribution and fixing the issues on the PR. This improvement will be released after completing the code reviews.

Best Regards,
Can

**Review by khushboo-singhvi — APPROVED** [review]

**Pull request merged**